src/classes/ADDR_Addresses_TEST.cls

```diff
@@ -998,6 +998,191 @@ public with sharing class ADDR_Addresses_TEST {
 
     /*********************************************************************************************************
     @description
+        Update Contact's mailing address and mark it as override, to cause creation of a new non-default Address 
+        record with the Mailing State value set as expected when State & Country Picklists are enabled.
+    verify:
+        Contact gets new Address
+        HH sticks with default Address
+        New Address created with State value set as expected
+    **********************************************************************************************************/
+    static testMethod void testAddressOverrideCreatesNewAddressWhenStateCodeIsChanged() {
+        if (!ADDR_Addresses_TDTM.isStateCountryPicklistsEnabled) {
+            return;
+        }
+
+        if (strTestOnly != '*' && strTestOnly != 'testAddressOverrideCreatesNewAddressWhenStateCodeIsChanged') {
+            return;
+        }
+
+        Integer hhSize = 1;
+        Integer contactSize = 1;
+        String state = 'California';
+        String stateCode = 'CA';
+
+        // This creates a default Address for each HH
+        createHHTestData(hhSize, contactSize);
+        System.assertEquals(1, getAddresses().size());
+
+        // Modify the Contact addresses directly, and set override
+        Contact con = listConT[0];
+        con.put('MailingStateCode', stateCode);
+        con.is_Address_Override__c = true;
+
+        Test.startTest();
+        update con;
+        Test.stopTest();
+
+        // Verify a new Address record was created
+        List<Address__c> retrievedAddresses = getAddresses();
+        System.assertEquals(2, retrievedAddresses.size());
+
+        // Find the default Address
+        Address__c defaultAddress = null;
+        Address__c newAddress = null;
+        for (Address__c eachAddress : retrievedAddresses) {
+            if (eachAddress.Default_Address__c) {
+                defaultAddress = eachAddress;
+            } else {
+                newAddress = eachAddress;
+            }
+        }
+
+        // Verify the Account's Billing Address info matches the Default Address record
+        assertEquals(listAddrT[0], defaultAddress);
+
+        // Verify the Contact's Address was updated as expected and that the Household and Contact Addresses are different.
+        List<Account> retrievedAccounts = Database.query(buildAccountSoql());
+        Map<Id, Account> accountIdToAccount = new Map<Id, Account>(retrievedAccounts);
+        List<Contact> listCon = Database.query(buildContactSoql());
+
+        System.assertEquals(state, listCon[0].MailingState);
+        System.assertEquals(true, listCon[0].is_Address_Override__c);
+        System.assertNotEquals(null, listCon[0].Current_Address__c);
+
+        Account eachContactAccount = accountIdToAccount.get(listCon[0].AccountId);
```

Comment:
Suggestion: since we only have one Account, we can call it `actualAccount` or just `account`.